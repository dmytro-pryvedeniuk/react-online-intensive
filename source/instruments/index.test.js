import {sum, delay, getUniqueID} from './';

describe('Instruments:', () => {
    test('sum should be a function', () => {
        expect(sum).toBeInstanceOf(Function);
    });
    
    test('sum should throw when called with non-number as second argument', () => {
        expect(() => sum(1, 'X')).toThrow();
    });

    test('sum should throw when called with non-number as first argument', () => {
        expect(() => sum('X', 2)).toThrow();
    });

    test('sum should sum two numbers', () => {
        expect(sum(1, 2)).toBe(3);
        expect(sum(7, 10)).toMatchSnapshot();
    });

    test('delay should return a resolved promise', async () => {
        await expect(delay(2)).resolves.toBe(2);
    });

    test('getUniqueID should be a function', () => {
        expect(getUniqueID).toBeInstanceOf(Function);
    });

    test('getUniqueID should throw when called with non-number', () => {
        expect(() => getUniqueID("X")).toThrow();
    });

    test('getUniqueID should return string with desired length', () => {
        expect(typeof getUniqueID()).toBe('string');
        expect(getUniqueID(5)).toHaveLength(5);
        expect(getUniqueID(13)).toHaveLength(13);
    });
})

