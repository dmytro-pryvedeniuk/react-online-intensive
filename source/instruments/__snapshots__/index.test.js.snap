// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instruments: sum should sum two numbers 1`] = `17`;
